### ISSUE TITLE
test_openFileDescriptors should pass the environment to the created process

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5442]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)| @therve reported|
|-|-|
|Trac ID|trac#5442|
|Type|defect|
|Created|2011-12-29 18:52:52Z|
|Branch|https://github.com/twisted/twisted/tree/test-openfd-env-5442|

If the test setup uses PYTHONPATH (for example to set the location of zope.interface), test_openFileDescriptors fails because the environment is not forwarded.

<details><summary>Searchable metadata</summary>

```
trac-id__5442 5442
type__defect defect
reporter__therve therve
priority__low low
milestone__ 
branch__branches_test_openfd_env_5442 branches/test-openfd-env-5442
branch_author__therve therve
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1325184772000000 1325184772000000
changetime__1376241992000000 1376241992000000
version__None None
owner__therve therve

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_1"></a>|@therve commented|
|-|-|

(In [39121]) Branching to 'test-openfd-env-5442'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @therve|
|-|-|

Does the environment need to be copied?  I think that `spawnProcess` should promise not to mutate its argument.

Otherwise, looks fine.  Please merge, with or without a fix for the copying, as you think is appropriate.

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_4"></a>|@therve set status to `closed`|
|-|-|

(In [39566]) Merge test-openfd-env-5442

Author: therve
Reviewer: exarkun
Fixes: [#5442](https://github.com/twisted/twisted/issues/5442)

Pass the environment in test_openFileDescriptors when creating a process to
make it work when PYTHONPATH is customized.
